syntax  = "proto3";

package arrus.proto;

import "io/proto/common/IntervalInteger.proto";

message ProbeAdapterModel {

  message Id {
    string manufacturer = 1;
    string name = 2;
  }

  message ChannelMappingRegion {
    uint32 us4oem = 1;
    repeated uint32 channels = 2;
    // Closed interval of channel numbers [first, last]
    IntervalInteger region = 3;
  }

  message ChannelMapping {
    repeated uint32 us4oems = 1;
    repeated uint32 channels = 2;
  }

  Id id = 1;
  uint32 n_channels = 2;
  ChannelMapping channel_mapping = 3;
  repeated ChannelMappingRegion channel_mapping_regions = 4;
}